projects/packages/jetpack-mu-wpcom/tests/php/features/launchpad/class-launchpad-task-definitions-test.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Test class for Launchpad_Task_Lists.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+/**
+ * Test class for Launchpad_Task_Lists.
+ *
+ * @coversDefaultClass Launchpad_Task_Lists
+ */
+class Launchpad_Task_Definitions_Test extends \WorDBless\BaseTestCase {
+	/**
+	 * Set up.
+	 */
+	public function set_up() {
+		parent::set_up();
+
+		add_filter(
+			'wpcom_launchpad_extended_task_definitions',
+			function () {
+				return array(
+					'test_task1'            => array(),
+					'test_task2'            => array(),
+					'test_task3'            => array(),
+					'test_task_with_id_map' => array(
+						'id_map' => 'test_task_id_map',
+					),
+				);
+			}
+		);
+	}
+
+	/**
+	 * Tests wether the wpcom_mark_launchpad_task_complete works correctly.
+	 */
+	public function test_wpcom_mark_launchpad_task_complete() {
+		wpcom_mark_launchpad_task_complete( 'test_task1' );
+		$options = get_option( 'launchpad_checklist_tasks_statuses' );
+		$this->assertTrue( isset( $options['test_task1'] ) );
+		$this->assertTrue( $options['test_task1'] );
+
+		wpcom_mark_launchpad_task_complete(
+			array(
+				'test_task2',
+				'test_task3',
+			)
+		);
+		$options = get_option( 'launchpad_checklist_tasks_statuses' );
+		$this->assertTrue( isset( $options['test_task2'] ) && $options['test_task2'] );
+		$this->assertTrue( isset( $options['test_task3'] ) && $options['test_task3'] );
+
+		$this->assertTrue( count( $options ) === 3 );
+	}
+
+	/**
+	 * Tests wether the wpcom_mark_launchpad_task_incomplete works correctly.
+	 */
+	public function test_wpcom_mark_launchpad_task_incomplete() {
+		wpcom_mark_launchpad_task_incomplete( 'test_task1' );
+		$options = get_option( 'launchpad_checklist_tasks_statuses' );
+		$this->assertTrue( isset( $options['test_task1'] ) && ! $options['test_task1'] );
+
+		wpcom_mark_launchpad_task_incomplete(
+			array(
+				'test_task2',
+				'test_task3',
+			)
+		);
+		$options = get_option( 'launchpad_checklist_tasks_statuses' );
+		$this->assertTrue( isset( $options['test_task2'] ) && ! $options['test_task2'] );
+		$this->assertTrue( isset( $options['test_task3'] ) && ! $options['test_task3'] );
```

Comment:
I thought the existing behaviour is that we only set values in the task list to be `true`, and don't set any falsey values.

Reply:
Yes, I think we shouldn't bloat the options data if we can avoid it, as the options data is aggressively cached, and we don't need the falsey values pushed into both the options table and the cache.